src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Jint.Native;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Int256;
+using Nethermind.TxPool;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPrice
+    {
+        private static UInt256? _lastPrice = null; //will this be okay if it is static?
+        private static Block? _lastHeadBlock = null;
+        private readonly IBlockFinder _blockFinder;
+        public GasPrice(IBlockFinder blockFinder)
+        {
+            _blockFinder = blockFinder;
```

Comment:
verify nullability as in other places in code, read about fail fast
I suggest to read the Clean Code book (C# style)